packages/runtimes/webextension/src/WebExtensionRuntime.js

```diff
@@ -50,16 +52,30 @@ export default (new Runtime({
 
       // Add autoreload
       if (bundle === firstInsertableBundle) {
-        return {
-          filePath: __filename,
-          code:
-            `var HMR_HOST = ${JSON.stringify(
-              options.hmrOptions?.host ?? 'localhost',
-            )};` +
-            `var HMR_PORT = '${options.hmrOptions?.port ?? ''}';` +
-            AUTORELOAD_BG,
-          isEntry: true,
-        };
+        return [
+          {
+            filePath: __filename,
+            code: AUTORELOAD_BG,
+            isEntry: true,
+          },
+          {
+            filePath: __filename,
+            // cache bust on non-asset manifest.json changes
```

Comment:
Why exactly are all of these caching workaround necessary here (this and `invalidateOnBuild` above)?

Reply:
This seems to be the only way to make the extension reload (i.e. have an HMR update sent out) when the `manifest.json` changes. I tried many other strategies but none seemed to work.